compiler/src/dotty/tools/dotc/core/Definitions.scala

```diff
@@ -1750,26 +1750,26 @@ class Definitions {
   @tu lazy val Tuple2SpecializedParamClasses: PerRun[Set[Symbol]] = new PerRun(Tuple2SpecializedParamTypes.map(_.symbol))
 
   // Specialized type parameters defined for scala.Function{0,1,2}.
-  @tu lazy val Function1SpecializedParamTypes: collection.Set[TypeRef] =
-    Set(IntType, LongType, FloatType, DoubleType)
-  @tu lazy val Function2SpecializedParamTypes: collection.Set[TypeRef] =
-    Set(IntType, LongType, DoubleType)
-  @tu lazy val Function0SpecializedReturnTypes: collection.Set[TypeRef] =
-    ScalaNumericValueTypeList.toSet + UnitType + BooleanType
-  @tu lazy val Function1SpecializedReturnTypes: collection.Set[TypeRef] =
-    Set(UnitType, BooleanType, IntType, FloatType, LongType, DoubleType)
-  @tu lazy val Function2SpecializedReturnTypes: collection.Set[TypeRef] =
+  @tu lazy val Function1SpecializedParamTypes: List[TypeRef] =
+    List(IntType, LongType, FloatType, DoubleType)
+  @tu lazy val Function2SpecializedParamTypes: List[TypeRef] =
+    List(IntType, LongType, DoubleType)
+  @tu lazy val Function0SpecializedReturnTypes: List[TypeRef] =
+    ScalaNumericValueTypeList :+ UnitType :+ BooleanType
+  @tu lazy val Function1SpecializedReturnTypes: List[TypeRef] =
+    List(UnitType, BooleanType, IntType, FloatType, LongType, DoubleType)
+  @tu lazy val Function2SpecializedReturnTypes: List[TypeRef] =
     Function1SpecializedReturnTypes
 
-  @tu lazy val Function1SpecializedParamClasses: PerRun[collection.Set[Symbol]] =
+  @tu lazy val Function1SpecializedParamClasses: PerRun[List[Symbol]] =
```

Comment:
We do rely on a fast `contains` operation on some of these.

Maybe we could type them as `collection.Set` and implement them with `collection.mutable.LinkedHashSet`.

Reply:
How large can those sets grow? Which is the largest one?

Reply:
Those can indeed get relatively large, like I think 216 elements for `Function2SpecializedApplyNames`. I changed those back to `Set`s, AFAICS now lists are used for iterating over and sets are used for `contains` checks. I'm not so familiar with `PerRun` though, does that look OK?